image_captioning/models/encoder/encoder_test.py

```diff
@@ -0,0 +1,48 @@
+"""
+Simple test code for the Encoder module
+"""
+import random
+import unittest
+
+import tensorflow as tf
+
+from image_captioning.models.encoder.encoder import Encoder
+
+
+class EncoderTest(unittest.TestCase):
+    """Test Encoder model
+    """
+
+    def test_inception_v3_encoder_output_shape(self):
+        """Test Inception V3 Encoder's output shape after call().
+        """
+        batch_size = random.randint(1, 10)
+        embedding_dim = random.randint(1, 64)
```

Comment:
nit: It is usually a bad idea to rely unit tests based on randomized inputs. This may result in spontaneous / flaky failures that are very confusing as the number of tests increase. Unit tests should be as deterministic as possible.

To test with multiple different inputs consider [parameterized testing](https://stackoverflow.com/a/34094) instead. No need to fix this now, though.

Reply:
Gotcha :)